pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -532,34 +532,36 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
       properties.setProperty(getKeyFor(column, DATETIME_GRANULARITY), dateTimeFieldSpec.getGranularity());
     }
 
-    String minValue = columnIndexCreationInfo.getMin().toString();
-    String maxValue = columnIndexCreationInfo.getMax().toString();
+    // NOTE: Min/max could be null for real-time aggregate metrics.
+    Object min = columnIndexCreationInfo.getMin();
```

Comment:
Without doing toString is there any impact on non null values?

Reply:
It is safest to put string instead of the object itself.
E.g. the following statements will throw exception (although if the save the properties and load again, everything will be loaded as string)
```
properties.setProperty("a", 1);
properties.getString("a");
```